sdk/search/search/src/searchIndexClient.ts

```diff
@@ -266,14 +278,19 @@ export class SearchIndexClient<T> {
     options: SearchOptions<Fields> = {},
     settings: ListSearchResultsPageSettings = {}
   ): AsyncIterableIterator<SearchDocumentsPageResult<Pick<T, Fields>>> {
-    let result = await this.searchDocuments<Fields>(options, settings.nextPageParameters);
+    const decodedContinuation = this.decodeContinuationToken(settings.continuationToken);
+    let result = await this.searchDocuments<Fields>(
+      options,
+      decodedContinuation?.nextPageParameters
+    );
 
     yield result;
 
     // Technically, we should also leverage nextLink, but the generated code
     // doesn't support this yet.
-    while (result.nextPageParameters) {
-      result = await this.searchDocuments(options, result.nextPageParameters);
+    while (result.continuationToken) {
+      const decodedContinuation = this.decodeContinuationToken(settings.continuationToken);
+      result = await this.searchDocuments(options, decodedContinuation?.nextPageParameters);
```

Comment:
Eh, kinda. Part of the trouble here was my private helper gets its result exposed in `byPage` so I'd have to be care to convert it there so I don't leak the type with nextParams out.

Since server pagination is so rare/uncommon I'm not too worried about hurting perf. Also I'd like to avoid getters since all of our result objects are "just data" and don't have magic/methods that don't serialize through JSON.